nixos/default.nix

```diff
@@ -65,6 +66,15 @@ in {
         '';
       };
 
+      sharedModules = mkOption {
+        type = types.listOf types.path;
```

Comment:
It should also be possible to use functions, not just paths.
I think the only way to type this is `types.listOf types.anything` and leave the checking to `imports`.